### PR TITLE
TestPeriodIndex.test_resample_empty_dataframe is failed when I pass kwargs to Resampler._groupby_and_aggregate

### LINKED ISSUE #22339: TestPeriodIndex.test_resample_empty_dataframe is failed when I pass kwargs to Resampler._groupby_and_aggregate
#### Code Sample, a copy-pastable example if possible

code diff to reproduce the issue

```
diff --git a/pandas/core/resample.py b/pandas/core/resample.py
index 4a3625b2d..c82382f96 100644
--- a/pandas/core/resample.py
+++ b/pandas/core/resample.py
@@ -1043,7 +1043,7 @@ class PeriodIndexResampler(DatetimeIndexResampler):

         if is_subperiod(ax.freq, self.freq):
             # Downsampling
-            return self._groupby_and_aggregate(how, grouper=self.grouper)
+            return self._groupby_and_aggregate(how, grouper=self.grouper, **kwargs)
         elif is_superperiod(ax.freq, self.freq):
             if how == 'ohlc':
                 # GH #13083
@@ -1051,7 +1051,7 @@ class PeriodIndexResampler(DatetimeIndexResampler):
                 # for pure aggregating/reducing methods
                 # OHLC reduces along the time dimension, but creates multiple
                 # values for each period -> handle by _groupby_and_aggregate()
-                return self._groupby_and_aggregate(how, grouper=self.grouper)
+                return self._groupby_and_aggregate(how, grouper=self.grouper, **kwargs)
             return self.asfreq()
         elif ax.freq == self.freq:
             return self.asfreq()
```
#### Problem description

During work on the solution of #15023 I found that when I pass `kwargs` to `_groupby_and_aggregate` the `TestPeriodIndex::test_resample_empty_dataframe` is failed.

@jreback suggested to create a separate issue for that [here](https://github.com/pandas-dev/pandas/pull/22304#discussion_r209901909)

trace:

```

self = <pandas.core.groupby.generic.DataFrameGroupBy object at 0x106674198>, func = 'var', args = (), kwargs = {'ddof': 1}

    @Appender(_apply_docs['template']
              .format(input="dataframe",
                      examples=_apply_docs['dataframe_examples']))
    def apply(self, func, *args, **kwargs):

        func = self._is_builtin_func(func)

        # this is needed so we don't try and wrap strings. If we could
        # resolve functions to their callable functions prior, this
        # wouldn't be needed
        if args or kwargs:
            if callable(func):

                @wraps(func)
                def f(g):
                    with np.errstate(all='ignore'):
                        return func(g, *args, **kwargs)
            else:
>               raise ValueError('func must be a callable if args or '
                                 'kwargs are supplied')
E               ValueError: func must be a callable if args or kwargs are supplied
```

#### Expected Output

`TestPeriodIndex::test_resample_empty_dataframe` shouldn't be failed cause I correctly passing `kwargs` to `Resampler._groupby_and_aggregate`.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: 1ea9664db77481eccf4fc9672ff9ac8283435f58
python: 3.5.3.candidate.1
python-bits: 64
OS: Darwin
OS-release: 16.7.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: en_US.UTF-8
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.24.0.dev0+465.g1ea9664db.dirty
pytest: 3.3.2
pip: 18.0
setuptools: 33.1.1
Cython: 0.28.4
numpy: 1.12.0
scipy: None
pyarrow: None
xarray: None
IPython: 5.2.2
sphinx: 1.6.6
patsy: None
dateutil: 2.6.0
pytz: 2016.10
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 2.0.0
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: 0.9999999
sqlalchemy: None
pymysql: None
psycopg2: None
jinja2: 2.9.5
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None

</details>

### PR DESCRIPTION
**Pull request by discort**
- [x] closes #22339
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

`test_resample.py::TestPeriodIndex::test_resample_empty_dataframe` triggers new changes

### CODE REVIEW

**Comment by discort** [comment]
@jreback 

I investigated `test_resample.py::TestPeriodIndex::test_resample_empty_dataframe` test and found that `Grouby.var` may fail when `ddof = 1`. Other methods have `_python_agg_general` in case of exception, so I added the same behaviour for `var`.

Let me know if it's acceptable solution.

**Comment by WillAyd** [comment]
This needs tests (should be created first)

**Review by WillAyd — CHANGES_REQUESTED** [review]
This needs tests (should be created first)

**Comment by discort** [comment]
@WillAyd 

After adding `kwargs` to `_groupby_and_aggregate`, changes are triggered  this test `test_resample.py::TestPeriodIndex::test_resample_empty_dataframe`.

**Comment by discort** [comment]
@WillAyd 

> Hmm not sure I completely follow - was this test failing before?

This test is failed using following changes.

```
diff --git a/pandas/core/resample.py b/pandas/core/resample.py
index 4a3625b2d..c82382f96 100644
--- a/pandas/core/resample.py
+++ b/pandas/core/resample.py
@@ -1043,7 +1043,7 @@ class PeriodIndexResampler(DatetimeIndexResampler):

         if is_subperiod(ax.freq, self.freq):
             # Downsampling
-            return self._groupby_and_aggregate(how, grouper=self.grouper)
+            return self._groupby_and_aggregate(how, grouper=self.grouper, **kwargs)
         elif is_superperiod(ax.freq, self.freq):
             if how == 'ohlc':
                 # GH #13083
@@ -1051,7 +1051,7 @@ class PeriodIndexResampler(DatetimeIndexResampler):
                 # for pure aggregating/reducing methods
                 # OHLC reduces along the time dimension, but creates multiple
                 # values for each period -> handle by _groupby_and_aggregate()
-                return self._groupby_and_aggregate(how, grouper=self.grouper)
+                return self._groupby_and_aggregate(how, grouper=self.grouper, **kwargs)
             return self.asfreq()
         elif ax.freq == self.freq:
             return self.asfreq()
```

**Comment by WillAyd** [comment]
How is this visible from the end user perspective? The test case you provided works fine on master so asserting that it continues to work doesn't measure much. 

However this is visible to an end user you should set up a test case to cover that

**Comment by discort** [comment]
@WillAyd

Just wanted to make sure that we are on the same page. @jreback suggested to create a new issue for this case [here](https://github.com/pandas-dev/pandas/pull/22304#discussion_r209901909). So, I created #22339 and this PR to close my original PR #22304, it's my goal at the moment.

I can make all changes within PR #22304 if it looks fine to you.

**Comment by WillAyd** [comment]
A separate PR is fine. Whatever you were planning to xfail in the other PR should be included as a test here

**Comment by discort** [comment]
@WillAyd 

There is already test that covers new changes: `TestPeriodIndex.test_resample_empty_dataframe`. I can create a separate test for `var`, for instance: `test_resample_empty_dataframe_var` which would duplicate the logic from `test_resample_empty_dataframe`. Are you okay with this approach?

**Comment by discort** [comment]
follow up ^. It would be more explicit

File: pandas/core/groupby/groupby.py [hunk 1]
```diff
@@ -1168,7 +1168,12 @@ def var(self, ddof=1, *args, **kwargs):
         """
         nv.validate_groupby_func('var', args, kwargs)
         if ddof == 1:
-            return self._cython_agg_general('var', **kwargs)
+            try:
+                return self._cython_agg_general('var', **kwargs)
+            except Exception:
```
Comment (jreback) [inline pandas/core/groupby/groupby.py]: can you make this exception more specific

**Review by jreback — CHANGES_REQUESTED** [review]
@discort we haven't merged that other PR yet, so you can build on top of it is probably best. IOW, in #22304 xfail the test and in this PR remove the xfail. We want to be sure things are tested.

**Comment by discort** [comment]
Will fix within https://github.com/pandas-dev/pandas/pull/22304

see [here](https://github.com/pandas-dev/pandas/pull/22304#discussion_r210545908)

**Pull request closed without merging**